lib/Language/Slicer/Desugar.hs

```diff
@@ -165,12 +173,33 @@ desugarM (A.Seq e1 e2)
                                     ++ " but it shold have unit type.")
          (e2', t2) <- desugarM e2
          return (ESeq e1' e2', t2)
+-- Exceptions
```

Comment:
In the "standard" (ML-like) treatment of exceptions, there is a special (extensible) datatype for exceptions, and anything we raise has to have that type.  Likewise, when we catch an exception, we can assume the exception has that type (and maybe pattern match on it).  This approach is somewhat different.